src/telemetry/activator.ts

```diff
@@ -19,41 +19,64 @@
 
 import { BallerinaExtension, ExtendedLangClient } from "src/core";
 import { window } from "vscode";
-import { getTelemetryProperties, TM_ERROR_LANG_SERVER, TM_EVENT_LANG_SERVER } from ".";
+import { getTelemetryProperties, TM_ERROR_LANG_SERVER, TM_EVENT_LANG_SERVER, TM_FEATURE_USAGE_LANG_SERVER } from ".";
+import { TM_EVENT_TYPE_ERROR, TM_EVENT_TYPE_FEATURE_USAGE } from "./events";
 
 export function activate(ballerinaExtInstance: BallerinaExtension) {
     const reporter = ballerinaExtInstance.telemetryReporter;
     const langClient = <ExtendedLangClient>ballerinaExtInstance.langClient;
 
     // Start listening telemtry events from language server
-    ballerinaExtInstance.onReady().then(() => {
-        langClient.onNotification('telemetry/event', (event: LSErrorTelemetryEvent | LSTelemetryEvent) => {
-            if ('errorMessage' in event) {
-                const errorEvent: LSErrorTelemetryEvent = event;
-                const props: { [key: string]: string; } = getTelemetryProperties(ballerinaExtInstance, event.component, event.message)
-                props["ballerina.langserver.error.stacktrace"] = errorEvent.errorStackTrace;
-                props["ballerina.langserver.error.message"] = errorEvent.errorMessage;
-                reporter.sendTelemetryEvent(TM_ERROR_LANG_SERVER, props);
-            } else {
-                reporter.sendTelemetryEvent(TM_EVENT_LANG_SERVER, getTelemetryProperties(ballerinaExtInstance, event.component, event.message));
-            }
-        });
-    })
+    ballerinaExtInstance.onReady()
+        .then(() => {
+            langClient.onNotification('telemetry/event', (event: LSTelemetryEvent) => {
+                let props: { [key: string]: string; };
+                switch (event.type) {
+                    case TM_EVENT_TYPE_ERROR:
+                        const errorEvent: LSErrorTelemetryEvent = <LSErrorTelemetryEvent>event;
+                        props = getTelemetryProperties(ballerinaExtInstance, event.component, errorEvent.message);
+                        props["ballerina.langserver.error.description"] = errorEvent.message;
+                        props["ballerina.langserver.error.stacktrace"] = errorEvent.errorStackTrace;
+                        props["ballerina.langserver.error.message"] = errorEvent.errorMessage;
+                        props["ballerina.langserver.event"] = JSON.stringify(event);
```

Comment:
Do we need the full event as a property? We have already set the properties separately.

Reply:
Suppose if we update the LS side to send more properties on an event, and updated the plugin to convert them here and send telemetry events. What if the user did not update the vscode plugin (which is highly possible)? We will lose those newly added properties. This way, we will have backward compatibility from data point of view. WDYT?

This had the drawback of the amount of data being send become large.

Reply:
Thanks! Removed this as discussed offline.